src/backends/unix/system.c

```diff
@@ -575,18 +575,22 @@ void
 Sys_RemoveDir(const char *path)
 {
 	char filepath[MAX_OSPATH];
-	DIR *directory = opendir(path);
 	struct dirent *file;
+	DIR *directory;
 
 	if (Sys_IsDir(path))
 	{
-		while ((file = readdir(directory)) != NULL)
+		directory = opendir(path);
+		if (directory)
 		{
-			snprintf(filepath, MAX_OSPATH, "%s/%s", path, file->d_name);
-			Sys_Remove(filepath);
-		}
+			while ((file = readdir(directory)) != NULL)
+			{
+				snprintf(filepath, MAX_OSPATH, "%s/%s", path, file->d_name);
+				Sys_Remove(filepath);
+			}
 
-		closedir(directory);
+			closedir(directory);
```

Comment:
This one should be in the `if (directory)` scope. If we can't open the directory, we likely can't remove it. And without having a look into the directory we cannot be sure that it's empty.